compiler/rustc_trait_selection/src/traits/vtable.rs

```diff
@@ -129,7 +131,8 @@ fn prepare_vtable_segments_inner<'tcx, T>(
                 .iter_identity_copied()
                 .filter_map(move |(pred, _)| {
                     Some(
-                        tcx.instantiate_bound_regions_with_erased(
+                        tcx.normalize_erasing_late_bound_regions(
+                            typing_env,
```

Comment:
it feels very suspicious to even theoretically support computing vtables in generic code. After all two types may only be unifiable after instantiation.

Should we instead change the entry-points to assert that the typing_env is empty and the trait ref concrete and otherwise bail/ICE?

Reply:
yeah but we will then either need to remove rustc_dump_vtable or rework it -- it's only there to support that, since all other callsites use the post mono typing env

Reply:
> yeah but we will then either need to remove rustc_dump_vtable or rework it 

hmm, move it to only be used on non-generic type aliases? or remove, idk